File: src/main/java/ch/leadrian/samp/kamp/textkeygen/TextKeysGeneratorPluginExtension.java
package ch.leadrian.samp.kamp.textkeygen;

import java.util.Set;

public class TextKeysGeneratorPluginExtension {

    private Object resourcesDirectory;
    private Set<String> packageNames;

    public Object getResourcesDirectory() {
        return resourcesDirectory;
    }

    public void setResourcesDirectory(Object resourcesDirectory) {
        this.resourcesDirectory = resourcesDirectory;
    }

    public Set<String> getPackageNames() {
        return packageNames;
    }

    public void setPackageNames(Set<String> packageNames) {
        this.packageNames = packageNames;
    }
}

File: src/main/kotlin/ch/leadrian/samp/kamp/gradle/plugin/textkeygen/TextKeysGeneratorPluginExtension.kt
package ch.leadrian.samp.kamp.gradle.plugin.textkeygen

import java.nio.charset.Charset
import java.nio.charset.StandardCharsets
import java.util.*

open class TextKeysGeneratorPluginExtension {

    lateinit var outputDirectory: Any

    lateinit var resourcesDirectory: Any

    var packageNames: Set<String> = HashSet()

    fun packageName(value: String) {
        packageNames += value
    }

    fun packageNames(vararg values: String) {
        packageNames += values
    }

    var charset: Charset = StandardCharsets.ISO_8859_1
}
